scripts/chandra/chandra.py

```diff
@@ -0,0 +1,209 @@
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = "https://cxc.cfa.harvard.edu/csc/"
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = ""
+
+_VERSION = "2.1.0"
+
+# Full data model here:
+# https://desidatamodel.readthedocs.io/en/latest/DESI_SPECTRO_REDUX/SPECPROD/zcatalog/zpix-SURVEY-PROGRAM.html
+
+#_BOOL_FEATURES = [
+#    "ZWARN"
+#]
+
+_FLOAT_FEATURES = [
+    "flux_aper_b",
+    "flux_bb_aper_b",
+    "flux_significance_b",
+    "hard_hm",
+    "hard_hs",
+    "hard_ms",
+    "var_index_b",
+    "var_prob_b",
+    
+]
+
+class CHANDRA(datasets.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    VERSION = _VERSION
+    
+    
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="csc21_spectra",
+            version=VERSION,
+            data_files=DataFilesPatternsDict.from_patterns(
+                {"train": ["./output_files/parent*.hdf5"]}
+            ),
+            description="X-Ray spectral data.",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "csc21_spectra"
+
+    #_spectrum_length = 7781
+
+    @classmethod
+    def _info(self):
+        """Defines the features available in this dataset."""
+        # Starting with all features common to image datasets
+        features = {
+            "spectrum": Sequence({
+                "ene_center_bin": Value(dtype="float32"),
+                "ene_high_bin": Value(dtype="float32"),
+                "ene_low_bin":  Value(dtype="float32"),
+                "flux": Value(dtype="float32"),
+                "flux_error": Value(dtype="float32"),
+            }, )
+        }
+
+        # Adding all values from the catalog
+        for f in _FLOAT_FEATURES:
+            features[f] = Value("float32")
+
+        # Adding all boolean flags
+        #for f in _BOOL_FEATURES:
+        #    features[f] = Value("bool")
+
+        features["name"] = Value("string")
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=Features(features),
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _generate_examples(self, files, object_ids=None):
+        """Yields examples as (key, example) tuples."""
+        for j, file in enumerate(itertools.chain.from_iterable(files)):
+            with h5py.File(file, "r") as data:
+                if object_ids is not None:
+                    keys = object_ids[j]
+                else:
+                    object_ids = [x.decode('utf-8') + '_' + str(y) + '_' + str(z) for x, y, z in zip(data["name"][:], data["obsid"][:], data["obi"][:])]
```

Comment:
This can be written cleaner as:
```suggestion
                    object_ids = [f"{name.decode('utf-8')}_{obsid}_{obi}" for name, obsid, obi in zip(data["name"][:], data["obsid"][:], data["obi"][:])]
```